ahcore/writers.py

```diff
@@ -105,21 +145,30 @@ def init_writer(self, first_batch: GenericArray, h5file: h5py.File) -> None:
                 chunks=(1,),
             )
 
+        if self._color_profile:
+            h5file.create_dataset(
+                "color_profile", data=np.frombuffer(self._color_profile, dtype=np.uint8), dtype="uint8"
+            )
+
         # This only works when the mode is 'overflow' and in 'C' order.
         metadata = {
             "mpp": self._mpp,
-            "dtype": str(batch_dtype),
```

Comment:
Removing `dtype` from metadata causes the reader to fail in the `_open_file` function.

Reply:
Two further comments:
- Applying the ICC profiles during training works (tested with TCGA images and Openslide backend). 
- When using the CLI functionality of `tiling.py`, a type error in `data.py` is raised. Simply ignoring the pylinting with a comment is not enough. I believe adding `from __future__ import annotations` at the top would solve the problem.

Reply:
I also observe this.

Reply:
Should be fixed?

Reply:
Yes, thanks!